src/Statement/StatementList.php

```diff
@@ -105,6 +106,34 @@ public function addNewStatement( Snak $mainSnak, $qualifiers = null, $references
 		$this->addStatement( $statement );
 	}
 
+	public function removeStatement( Statement $statement ) {
```

Comment:
As far as I can tell, the equivalent method in `Claims` is actually never used. Removing a Statement by GUID is however something currently done, with this commit proving we have use for such a method: https://gerrit.wikimedia.org/r/#/c/204511/